packages/paste-website/src/pages/primitives/menu-primitive/index.mdx

```diff
@@ -0,0 +1,257 @@
+
+export const pageQuery = graphql`
+  {
+    allPastePrimitive(filter: {name: {eq: "@twilio-paste/menu-primitive"}}) {
+      edges {
+        node {
+          name
+          description
+          status
+          version
+        }
+      }
+    }
+    menuBarExample: file(
+      sourceInstanceName: {eq: "assets"}
+      relativePath: {eq: "images/menu-primitive-images/menubar.png"}
+    ) {
+      childImageSharp {
+        fluid(maxWidth: 500) {
+          ...GatsbyImageSharpFluid
+        }
+      }
+    }
+    mdx(fields: {slug: {eq: "/menu-primitive/"}}) {
+      headings {
+        depth
+        value
+      }
+    }
+  }
+`;
+
+<ComponentHeader
+  name="Menu Primitive"
+  categoryRoute={SidebarCategoryRoutes.PRIMITIVES}
+  githubUrl="https://github.com/twilio-labs/paste/tree/master/packages/paste-core/primitives/menu"
+  storybookUrl="https://twilio-labs.github.io/paste/?path=/story/primitives-menu--default"
+  data={props.data.allPastePrimitive.edges}
+/>
+
+---
+
+<contentwrapper>
+
+<TableOfContents headings={props.data.mdx.headings} />
+
+<content>
+
+## Guidelines
+
+### About Menu Primitive
+
+The menu primitive is an unstyled functional version of a menu. It can be used to build a component following the WAI-ARIA Menu pattern. Our [Menu](/components/menu) is built on top of this primitive.
+
+The purpose of providing these unstyled primitives is to cater for instances when the styled Menu provided by Paste, doesn’t meet the requirements needed to solve a unique or individual customer problem. At that point you are welcome to fallback to this functional primitive to roll your own styled menu while still providing a functional and accessible experience to your customers.
+
+This primitive should be used to compose all custom menus to ensure accessibility and upgrade paths.
+
+<Callout variant="warning">
+  <CalloutTitle as="h3">Warning</CalloutTitle>
+  <CalloutText>
+    We strongly suggest that all components built on top of this primitive get reviewed by the Design Systems team and
+    go through the UX Review process to ensure an excellent experience for our customers.
+  </CalloutText>
+</Callout>
+
+## Examples
+
+### Basic menu
+
+<LivePreview
+  scope={{
+    useMenuPrimitiveState,
+    MenuPrimitiveButton,
+    MenuPrimitive,
+    MenuPrimitiveItem,
+    MenuPrimitiveSeparator,
+  }}
+  noInline
+>
+  {defaultExample}
+</LivePreview>
+
+### Sub menu
+
+<LivePreview
+  scope={{
+    useMenuPrimitiveState,
+    MenuPrimitiveButton,
+    MenuPrimitive,
+    MenuPrimitiveItem,
+    MenuPrimitiveSeparator,
+  }}
+  noInline
+>
+  {subMenuExample}
+</LivePreview>
+
+### Custom styling
+
+You can provide custom styling to the primitive menu by utilizing the `as` prop on each component.
+
+Because these are not styled, rendering any of them _as_ a another component you can mix the functionlity of two components together. Styling from one, menu functionlity from the primitive component.
```

Comment:
```suggestion
The menu primitive does not come with any styling, and thus you could mix the functionality of it with another component by using the _as_ prop. By doing so, you can get styling from another component, and menu functionality from this primitive. 

Because these are not styled, rendering any of them _as_ another component you can mix the functionality of two components together. Styling from one, menu functionlity from the primitive component.
```